### PR TITLE
All HAMs use responseUnauthorized

### PR DESCRIPTION
**Pull request by Zech-Hein**
For https://github.com/OpenLiberty/open-liberty/issues/23326

### CODE REVIEW

**Comment by Zech-Hein** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.javaeesec_fat,com.ibm.ws.security.javaeesec_fat.2,com.ibm.ws.security.javaeesec_fat.3,com.ibm.ws.security.javaeesec_fat.4,com.ibm.ws.security.jaspic_fat,com.ibm.ws.security.jaspic.jakarta10_fat,io.openliberty.security.jakartasec.3.0.internal_fat, io.openliberty.security.jakartasec.3.0.internal_fat.config

File: dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/beans/CustomFormAuthenticationMechanism.java [hunk 1]
```diff
@@ -109,7 +106,7 @@ public void cleanSubject(HttpServletRequest request,
     private AuthenticationStatus handleFormLogin(@Sensitive Credential credential, HttpServletResponse rsp, Subject clientSubject,
                                                  HttpMessageContext httpMessageContext) throws AuthenticationException {
         AuthenticationStatus status = utils.handleAuthenticate(getCDI(), JavaEESecConstants.DEFAULT_REALM, credential, clientSubject, httpMessageContext);
-        int rspStatus;
+        int rspStatus = -1;
```
Comment (utle) [inline dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/beans/CustomFormAuthenticationMechanism.java:109]: rspStatus;

File: dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/beans/CustomFormAuthenticationMechanism.java [hunk 2]
```diff
@@ -119,10 +116,10 @@ private AuthenticationStatus handleFormLogin(@Sensitive Credential credential, H
             // set SC_OK, since if the target is not protected, it'll be processed.
             rspStatus = HttpServletResponse.SC_OK;
         } else {
-            rspStatus = HttpServletResponse.SC_UNAUTHORIZED;
+            httpMessageContext.responseUnauthorized();
             // TODO: Audit invalid user or password
```
Comment (utle) [inline dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/beans/CustomFormAuthenticationMechanism.java:120]: rspStatus = HttpServletResponse.SC_UNAUTHORIZED;

File: dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/beans/CustomFormAuthenticationMechanism.java [hunk 3]
```diff
@@ -119,10 +116,10 @@ private AuthenticationStatus handleFormLogin(@Sensitive Credential credential, H
...
         }
-        if (rsp != null) {
+        if (rsp != null && rspStatus != -1) {
```
Comment (utle) [inline dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/beans/CustomFormAuthenticationMechanism.java:122]: if (rsp != null)

**Review by utle — COMMENTED** [review]
int rspStatus;

**Review by utle — APPROVED** [review]

**Pull request merged**